src/storage/v2/vertex_accessor.cpp

```diff
@@ -371,20 +410,28 @@ Result<std::map<PropertyId, PropertyValue>> VertexAccessor::ClearProperties() {
 
   using ReturnType = decltype(vertex_->properties.Properties());
   std::optional<ReturnType> properties;
-  utils::AtomicMemoryBlock atomic_memory_block{
-      [storage = storage_, transaction = transaction_, vertex = vertex_, &properties]() {
-        properties.emplace(vertex->properties.Properties());
-        if (!properties.has_value()) {
-          return;
-        }
-        for (const auto &[property, value] : *properties) {
-          CreateAndLinkDelta(transaction, vertex, Delta::SetPropertyTag(), property, value);
-          storage->indices_.UpdateOnSetProperty(property, PropertyValue(), vertex, *transaction);
-          transaction->constraint_verification_info.RemovedProperty(vertex);
-          transaction->manyDeltasCache.Invalidate(vertex, property);
-        }
-        vertex->properties.ClearProperties();
-      }};
+  utils::AtomicMemoryBlock atomic_memory_block{[storage = storage_, transaction = transaction_, vertex = vertex_,
+                                                &properties]() {
+    properties.emplace(vertex->properties.Properties());
+    if (!properties.has_value()) {
+      return;
+    }
+    for (const auto &[property, value] : *properties) {
+      CreateAndLinkDelta(transaction, vertex, Delta::SetPropertyTag(), property, value);
+      storage->indices_.UpdateOnSetProperty(property, PropertyValue(), vertex, *transaction);
+      transaction->constraint_verification_info.RemovedProperty(vertex);
+      transaction->manyDeltasCache.Invalidate(vertex, property);
+    }
+
+    vertex->properties.ClearProperties();
+    if (flags::run_time::GetTextSearchEnabled()) {
+      for (const auto *index_context : storage->indices_.text_index_->GetApplicableTextIndices(vertex, storage)) {
+        auto search_input =
+            mgcxx_mock::text_search::SearchInput{.search_query = fmt::format("metadata.gid:{}", vertex->gid.AsInt())};
+        mgcxx_mock::text_search::Mock::delete_document(*index_context, search_input, true);
```

Comment:
What if there is nothing to delete, what happens then? Do we still need to call mgcxx_mock?

Reply:
This snippet is from the `ClearProperties` method, which is only ever called when replacing properties (`SetProperties` operator with `REPLACE`). Data in Tantivy is immutable, so the necessary step is to delete the existing document.